linera-core/src/client/mod.rs

```diff
@@ -2101,12 +2101,20 @@ where
         &self,
         operations: Vec<Operation>,
         blobs: Vec<Blob>,
+        guard: Option<OwnedMutexGuard<()>>,
```

Comment:
I guess the `Option` is just to allow the profiling to include taking the lock?  I think I'd rather it always did or always didn't, especially since there's no label in the metric to indicate whether we are taking the lock or not.

Reply:
It's for the case when we already took the lock in the outside context or not b/c this method is called in more than one place (and sometimes we have the lock and sometimes we don't).

Reply:
It's just so that I don't have to explicitly acquire the lock in the other call sites where we don't want the guard to live longer than the `execute_operation` call.

Reply:
Right, I understand that, but in that case it should be equivalent to taking the lock and passing the (owned) guard as an argument, which also gives the metric the semantics of always being after the lock acquisition.

If the code to acquire the lock is too long or we expect it to change, we can always factor it into a function.

Reply:
Good point, fixed in https://github.com/linera-io/linera-protocol/pull/3822/commits/9aaf71c13004e8780fa57607c01d992b7d1bb481.

Reply:
This is hopefully only temporary anyway: I will try (on the main branch) to remove the whole client mutex, but that will be a much bigger change.